paramiko/proxy.py

```diff
@@ -105,7 +105,8 @@ def recv(self, size):
             raise ProxyCommandFailure(' '.join(self.cmd), e.strerror)
 
     def close(self):
-        os.kill(self.process.pid, signal.SIGTERM)
+        self.process.kill()
```

Comment:
I found it's good to wrap this kill in a try block (see https://github.com/paramiko/paramiko/pull/962), as otherwise the program calling paramiko will traceback if something else killed the process.